cpp/collective/rendezvous/CMakeLists.txt

```diff
@@ -5,13 +5,16 @@ project(
   VERSION 0.0.1
   LANGUAGES CXX)
 
-set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
Only for windows C++17. Other platform remain C++ 20. And add a comment to explain why windows needs C++17.